README.md
# An Enum listing HTTP Statuses

[![Latest Version on Packagist](https://img.shields.io/packagist/v/zero-to-prod/http-status.svg?style=flat-square)](https://packagist.org/packages/zero-to-prod/http-status)
[![GitHub Tests Action Status](https://img.shields.io/github/workflow/status/zero-to-prod/http-status/run-tests?label=tests)](https://github.com/zero-to-prod/http-status/actions?query=workflow%3Arun-tests+branch%3Amain)
[![Total Downloads](https://img.shields.io/packagist/dt/zero-to-prod/http-status.svg?style=flat-square)](https://packagist.org/packages/zero-to-prod/http-status)

An Enum containing Http Statuses.

## Installation

You can install the package via composer:

```bash
composer require zero-to-prod/http-status
```

## Changelog

Please see [CHANGELOG](CHANGELOG.md) for more information on what has changed recently.

## Contributing

Please see [CONTRIBUTING](.github/CONTRIBUTING.md) for details.

## Security Vulnerabilities

Please review [our security policy](../../security/policy) on how to report security vulnerabilities.

## Credits

- [D.A.Smith](https://github.com/DavidSmith)

## License

The MIT License (MIT). Please see [License File](LICENSE.md) for more information.


CHANGELOG.md
# Changelog

All notable changes to `HttpStatus` will be documented in this file.

## 1.0.0 - 2022-01-08

- initial release


tests/Pest.php
<?php

use ZeroToProd\HttpStatus\Tests\TestCase;

uses(TestCase::class)->in(__DIR__);


tests/TestCase.php
<?php

namespace ZeroToProd\HttpStatus\Tests;

use Illuminate\Database\Eloquent\Factories\Factory;
use Orchestra\Testbench\TestCase as Orchestra;
use ZeroToProd\HttpStatus\HttpStatusServiceProvider;

class TestCase extends Orchestra
{
    protected function setUp(): void
    {
        parent::setUp();

        Factory::guessFactoryNamesUsing(
            fn (string $modelName) => 'ZeroToProd\\HttpStatus\\Database\\Factories\\'.class_basename($modelName).'Factory'
        );
    }

    protected function getPackageProviders($app)
    {
        return [
            HttpStatusServiceProvider::class,
        ];
    }

    public function getEnvironmentSetUp($app)
    {
        config()->set('database.default', 'testing');

        /*
        $migration = include __DIR__.'/../database/migrations/create_httpstatus_table.php.stub';
        $migration->up();
        */
    }
}


tests/HttpStatusTest.php
<?php

it('returns the correct status code for Continue', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::Continue->value)->toBe(100);
});
it('returns the correct status code for SwitchingProtocols', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::SwitchingProtocols->value)->toBe(101);
});
it('returns the correct status code for Processing', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::Processing->value)->toBe(102);
});
it('returns the correct status code for EarlyHints', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::EarlyHints->value)->toBe(103);
});
it('returns the correct status code for OK', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::OK->value)->toBe(200);
});
it('returns the correct status code for Created', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::Created->value)->toBe(201);
});
it('returns the correct status code for Accepted', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::Accepted->value)->toBe(202);
});
it('returns the correct status code for NonAuthoritativeInformation', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::NonAuthoritativeInformation->value)->toBe(203);
});
it('returns the correct status code for NoContent', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::NoContent->value)->toBe(204);
});
it('returns the correct status code for ResetContent', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::ResetContent->value)->toBe(205);
});
it('returns the correct status code for PartialContent', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::PartialContent->value)->toBe(206);
});
it('returns the correct status code for MultiStatus', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::MultiStatus->value)->toBe(207);
});
it('returns the correct status code for AlreadyReported', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::AlreadyReported->value)->toBe(208);
});
it('returns the correct status code for IMUsed', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::IMUsed->value)->toBe(226);
});
it('returns the correct status code for MultipleChoices', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::MultipleChoices->value)->toBe(300);
});
it('returns the correct status code for MovedPermanently', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::MovedPermanently->value)->toBe(301);
});
it('returns the correct status code for Found', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::Found->value)->toBe(302);
});
it('returns the correct status code for SeeOther', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::SeeOther->value)->toBe(303);
});
it('returns the correct status code for NotModified', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::NotModified->value)->toBe(304);
});
it('returns the correct status code for UseProxy', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::UseProxy->value)->toBe(305);
});
it('returns the correct status code for SwitchCase', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::SwitchCase->value)->toBe(306);
});
it('returns the correct status code for TemporaryRedirect', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::TemporaryRedirect->value)->toBe(307);
});
it('returns the correct status code for PermanentRedirect', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::PermanentRedirect->value)->toBe(308);
});
it('returns the correct status code for BadRequest', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::BadRequest->value)->toBe(400);
});
it('returns the correct status code for Unauthorized', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::Unauthorized->value)->toBe(401);
});
it('returns the correct status code for PaymentRequired', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::PaymentRequired->value)->toBe(402);
});
it('returns the correct status code for Forbidden', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::Forbidden->value)->toBe(403);
});
it('returns the correct status code for NotFound', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::NotFound->value)->toBe(404);
});
it('returns the correct status code for MethodNotAllowed', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::MethodNotAllowed->value)->toBe(405);
});
it('returns the correct status code for NotAcceptable', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::NotAcceptable->value)->toBe(406);
});
it('returns the correct status code for ProxyAuthenticationRequired', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::ProxyAuthenticationRequired->value)->toBe(407);
});
it('returns the correct status code for RequestTimeout', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::RequestTimeout->value)->toBe(408);
});
it('returns the correct status code for Conflict', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::Conflict->value)->toBe(409);
});
it('returns the correct status code for Gone', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::Gone->value)->toBe(410);
});
it('returns the correct status code for LengthRequired', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::LengthRequired->value)->toBe(411);
});
it('returns the correct status code for PreconditionFailed', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::PreconditionFailed->value)->toBe(412);
});
it('returns the correct status code for PayloadTooLarge', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::PayloadTooLarge->value)->toBe(413);
});
it('returns the correct status code for URITooLong', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::URITooLong->value)->toBe(414);
});
it('returns the correct status code for UnsupportedMediaType', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::UnsupportedMediaType->value)->toBe(415);
});
it('returns the correct status code for RangeNotSatisfiable', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::RangeNotSatisfiable->value)->toBe(416);
});
it('returns the correct status code for ExpectationFailed', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::ExpectationFailed->value)->toBe(417);
});
it('returns the correct status code for ImATeapot', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::ImATeapot->value)->toBe(418);
});
it('returns the correct status code for MisdirectedRequest', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::MisdirectedRequest->value)->toBe(421);
});
it('returns the correct status code for UnprocessableEntity', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::UnprocessableEntity->value)->toBe(422);
});
it('returns the correct status code for Locked', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::Locked->value)->toBe(423);
});
it('returns the correct status code for FailedDependency', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::FailedDependency->value)->toBe(424);
});
it('returns the correct status code for TooEarly', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::TooEarly->value)->toBe(425);
});
it('returns the correct status code for UpgradeRequired', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::UpgradeRequired->value)->toBe(426);
});
it('returns the correct status code for PreconditionRequired', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::PreconditionRequired->value)->toBe(428);
});
it('returns the correct status code for TooManyRequests', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::TooManyRequests->value)->toBe(429);
});
it('returns the correct status code for RequestHeaderFieldsTooLarge', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::RequestHeaderFieldsTooLarge->value)->toBe(431);
});
it('returns the correct status code for UnavailableForLegalReasons', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::UnavailableForLegalReasons->value)->toBe(451);
});
it('returns the correct status code for InternalServerError', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::InternalServerError->value)->toBe(500);
});
it('returns the correct status code for NotImplemented', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::NotImplemented->value)->toBe(501);
});
it('returns the correct status code for BadGateway', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::BadGateway->value)->toBe(502);
});
it('returns the correct status code for ServiceUnavailable', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::ServiceUnavailable->value)->toBe(503);
});
it('returns the correct status code for GatewayTimeout', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::GatewayTimeout->value)->toBe(504);
});
it('returns the correct status code for HTTPVersionNotSupported', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::HTTPVersionNotSupported->value)->toBe(505);
});
it('returns the correct status code for VariantAlsoNegotiates', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::VariantAlsoNegotiates->value)->toBe(506);
});
it('returns the correct status code for InsufficientStorage', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::InsufficientStorage->value)->toBe(507);
});
it('returns the correct status code for LoopDetected', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::LoopDetected->value)->toBe(508);
});
it('returns the correct status code for NotExtended', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::NotExtended->value)->toBe(510);
});
it('returns the correct status code for NetworkAuthenticationRequired', function () {
    expect(ZeroToProd\HttpStatus\HttpStatus::NetworkAuthenticationRequired->value)->toBe(511);
});


src/HttpStatus.php
<?php

namespace ZeroToProd\HttpStatus;

enum HttpStatus: int
{
    case Continue = 100;
    case SwitchingProtocols = 101;
    case Processing = 102;
    case EarlyHints = 103;
    case OK = 200;
    case Created = 201;
    case Accepted = 202;
    case NonAuthoritativeInformation = 203;
    case NoContent = 204;
    case ResetContent = 205;
    case PartialContent = 206;
    case MultiStatus = 207;
    case AlreadyReported = 208;
    case IMUsed = 226;
    case MultipleChoices = 300;
    case MovedPermanently = 301;
    case Found = 302;
    case SeeOther = 303;
    case NotModified = 304;
    case UseProxy = 305;
    case SwitchCase = 306;
    case TemporaryRedirect = 307;
    case PermanentRedirect = 308;
    case BadRequest = 400;
    case Unauthorized = 401;
    case PaymentRequired = 402;
    case Forbidden = 403;
    case NotFound = 404;
    case MethodNotAllowed = 405;
    case NotAcceptable = 406;
    case ProxyAuthenticationRequired = 407;
    case RequestTimeout = 408;
    case Conflict = 409;
    case Gone = 410;
    case LengthRequired = 411;
    case PreconditionFailed = 412;
    case PayloadTooLarge = 413;
    case URITooLong = 414;
    case UnsupportedMediaType = 415;
    case RangeNotSatisfiable = 416;
    case ExpectationFailed = 417;
    case ImATeapot = 418;
    case MisdirectedRequest = 421;
    case UnprocessableEntity = 422;
    case Locked = 423;
    case FailedDependency = 424;
    case TooEarly = 425;
    case UpgradeRequired = 426;
    case PreconditionRequired = 428;
    case TooManyRequests = 429;
    case RequestHeaderFieldsTooLarge = 431;
    case UnavailableForLegalReasons = 451;
    case InternalServerError = 500;
    case NotImplemented = 501;
    case BadGateway = 502;
    case ServiceUnavailable = 503;
    case GatewayTimeout = 504;
    case HTTPVersionNotSupported = 505;
    case VariantAlsoNegotiates = 506;
    case InsufficientStorage = 507;
    case LoopDetected = 508;
    case NotExtended = 510;
    case NetworkAuthenticationRequired = 511;
}
